### PR TITLE
property added to disable or enable role based scope validation for implicit grant

### PR DESCRIPTION
**Pull request by Buddhimah**
### Proposed changes in this pull request

Since the implicit grant did not validate the scopes before, there could be apps implemented to work with this behavior. Just like our playground app fails when this fix[1][2] is applied, those applications would also fail when the fix is applied. Therefore, I think we should add a property and provide the ability for the user to enable/disable this fix[1][2].

#### Configuration Changes

Following configuration should be added to  <IS_HOME>/repository/conf/identity/identity.xml under 
tag  <OAuth> to disable scope validation for Implicit and Authorization Code Grant  types when XACML Scope Validator is enabled. This should go as a documentation update. [3] 

`        <ScopeValidationEnabledForAuthzCodeAndImplicitGrant>false</ScopeValidationEnabledForAuthzCodeAndImplicitGrant>`

Currently this is added as a commented property to identity.xml. [4]

[1].[https://github.com/wso2-support/identity-inbound-auth-oauth/pull/423](https://github.com/wso2-support/identity-inbound-auth-oauth/pull/423)

[2].[wso2-extensions/identity-inbound-auth-oauth/pull/1123](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1123)

[3].https://docs.wso2.com/display/IS580/Validating+the+Scope+of+OAuth+Access+Tokens+using+XACML+Policies

[4].[https://github.com/wso2/carbon-identity-framework/pull/2242](https://github.com/wso2/carbon-identity-framework/pull/2242)

#### Related Issue

> wso2/product-is#5335

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/Oauth2ScopeUtils.java [hunk 1]
```diff
@@ -271,4 +273,26 @@ private static boolean iterateOAuth2ScopeValidators(OAuthAuthzReqMessageContext
...
         return true;
     }
+
+    /**
+     * Read RoleBasedScopeValidation configurations for Implicit Grant from identity.xml file.
+     *
+     * @param xPath xpath of the config property.
+     * @return Config property value.
+     */
+    public static boolean readRoleBasedScopeValidationEnabledConfigValue(String xPath) {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/Oauth2ScopeUtils.java]: We usually get OAuth related configurations from OAuthServerConfiguration object. Shouldn't we do the same here as well?

Comment (Buddhimah) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/Oauth2ScopeUtils.java]: done

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java [hunk 2]
```diff
@@ -2706,6 +2715,10 @@ public void populateOAuthTokenIssuerMap() throws IdentityOAuth2Exception {
         }
     }
 
+    public boolean isScopeValidationEnabledForCodeAndImplicitGran() {
```
Comment (Abilashini) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: missing 't' at the end 
isScopeValidationEnabledForCodeAndImplicitGran -> isScopeValidationEnabledForCodeAndImplicitGrant

Comment (Abilashini) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: Let's add Java Doc

Comment (Buddhimah) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: done

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java [hunk 3]
```diff
@@ -290,6 +291,14 @@ private void buildOAuthServerConfiguration() {
         OMElement scopeValidatorsElem = oauthElem.getFirstChildWithName(
                 getQNameWithIdentityNS(ConfigElements.SCOPE_VALIDATORS));
 
+        //Get scopeValidationEnabledConfigValue
```
Comment (JKAUSHALYA) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: Please use the proper commenting style in all the places.

Comment (Buddhimah) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java]: Fix https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1294

**Pull request merged**